cdap-ui/app/cdap/components/PreviewData/RecordView/Navigator.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import IconSVG from 'components/IconSVG';
+import classnames from 'classnames';
+import withStyles, { WithStyles, StyleRules } from '@material-ui/core/styles/withStyles';
+import AbstractWidget from 'components/AbstractWidget';
+
+const styles = (theme): StyleRules => ({
+  root: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    padding: '5px 10px',
+    fontSize: '1.5rem',
+    borderBottom: `1px solid ${theme.palette.grey[400]}`,
+  },
+  btnIcon: {
+    padding: '10px',
+  },
+});
+
+interface IRecordNavigatorProps extends WithStyles<typeof styles> {
+  selectedRecord: number;
+  numRecords: number;
+  updateRecord: (e: React.ChangeEvent<HTMLSelectElement>) => void;
+  prevOperation: () => void;
+  nextOperation: () => void;
+}
+
+const RecordNavigatorBase: React.FC<IRecordNavigatorProps> = ({
+  classes,
+  selectedRecord,
+  numRecords,
+  updateRecord,
+  prevOperation,
+  nextOperation,
+}) => {
+  const prevDisabled = selectedRecord - 1 < 1;
+  const nextDisabled = selectedRecord + 1 > numRecords;
+
+  const selectOptions = [];
+  for (let i = 1; i <= numRecords; i++) {
+    const value = `Record ${i}`;
+    selectOptions.push({ value, label: value });
+  }
+
+  return (
+    <div className={classes.root}>
+      <span
+        className={classnames('nav-icon', { disabled: prevDisabled })}
+        onClick={!prevDisabled ? prevOperation : undefined}
+      >
+        <IconSVG name="icon-caret-left" />
+      </span>
+      <span>
+        <AbstractWidget
+          value={`Record ${selectedRecord}`}
+          type="select"
+          widgetProps={{ options: selectOptions }}
+          onChange={(e) => updateRecord(e)}
+        />
+      </span>
+      <span
+        className={classnames('nav-icon', { disabled: nextDisabled })}
+        onClick={!nextDisabled ? nextOperation : undefined}
+      >
+        <IconSVG name="icon-caret-right" />
+      </span>
```

Comment:
can you investigate to see if perhaps using material-ui button icons would be easy to style? this would prevent additional disabled styling and onClick handling

Reply:
Used material-ui IconButton component instead.